pkg/auth/oauth/external/gitlab/gitlab.go

```diff
@@ -0,0 +1,111 @@
+package gitlab
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/RangelReale/osincli"
+	"github.com/golang/glog"
+
+	authapi "github.com/openshift/origin/pkg/auth/api"
+	"github.com/openshift/origin/pkg/auth/oauth/external"
+)
+
+const (
+	gitlabAuthorizePath = "oauth/authorize"
+	gitlabTokenPath     = "oauth/token"
+	gitlabUserApiPath   = "api/v3/user"
```

Comment:
are there other versions that support user info? we'd like as broad support as possible... again, any chance of a link to documentation about which version(s) support getting user info?